src/lib/isMobilePhone.js

```diff
@@ -158,6 +158,8 @@ const phones = {
   'ar-YE': /^(((\+|00)9677|0?7)[0137]\d{7}|((\+|00)967|0)[1-7]\d{6})$/,
   'ar-EH': /^(\+?212|0)[\s\-]?(5288|5289)[\s\-]?\d{5}$/,
   'fa-AF': /^(\+93|0)?(2{1}[0-8]{1}|[3-5]{1}[0-4]{1})(\d{7})$/,
+  'gu-GU': /^(\+?1|1)?\s*671\s*\d{7}$/,
```

Comment:
I think the regex can be shortened to `/^(\+?1|1)?671\d{7}$/` to be more in line with the others. There are specific ranges in use, see https://www.area-codes.com/area-code/area-code-671.asp but I think we don't need to validate for each of them at this moment. I also could not find documentation yet on which phone numbers are for mobile usage so we also cannot validate those specifically. See also the notes from Google's libphonenumer; https://github.com/google/libphonenumber/blob/master/resources/metadata/1/comments.csv

I also think it's better to use  `en-GU` instead of `gu-GU`. See https://www.iso.org/obp/ui/#iso:code:3166:GU